tests/main/views/test_audits.py

```diff
@@ -50,8 +54,204 @@ def add_audit_events_with_db_object(self):
+                    ),
+                    (  # supplier_audit_event_params
+                        (1, AuditTypes.supplier_update, datetime(2010, 2, 6), None,),
+                        (0, AuditTypes.supplier_update, datetime(2010, 6, 6), None,),
+                        (1, AuditTypes.supplier_update, datetime(2010, 2, 6), None,),
+                    ),
+                    (  # series of req_cases for the above db scenario
+                        (
+                            {"earliest_for_each_object": "true"},
+                            (6, 7, 4, 3, 5,),
+                        ),
+                        (
+                            {"earliest_for_each_object": "true", "acknowledged": "false"},
+                            (6, 7, 0, 5,),
+                        ),
+                        (
+                            {"earliest_for_each_object": "true", "acknowledged": "true"},
+                            (4, 3,),
+                        ),
+                        (
+                            {"earliest_for_each_object": "true", "object-type": "suppliers"},
+                            (6, 7,),
+                        ),
+                        (
+                            {"earliest_for_each_object": "true", "audit-date": "2011-08-06"},
+                            (3,),
+                        ),
+                        (
+                            {"earliest_for_each_object": "false", "per_page": "100"},
+                            (6, 8, 7, 4, 0, 1, 3, 2, 5,),
+                        ),
+                        (
+                            {"per_page": "100"},
+                            (6, 8, 7, 4, 0, 1, 3, 2, 5,),
+                        ),
+                    ),
+                ),
+                (
+                    (  # service_audit_event_params
+                        (0, AuditTypes.update_service, datetime(2010, 8, 5), datetime(2011, 8, 2),),
+                        (0, AuditTypes.update_service, datetime(2010, 8, 4), datetime(2011, 8, 2),),
+                    ),
+                    (  # supplier_audit_event_params
+                        (0, AuditTypes.supplier_update, datetime(2010, 6, 6), None,),
+                        (1, AuditTypes.supplier_update, datetime(2010, 2, 1), None,),
+                        (1, AuditTypes.supplier_update, datetime(2010, 2, 6), None,),
+                        (0, AuditTypes.supplier_update, datetime(2010, 2, 7), None,),
+                        (3, AuditTypes.supplier_update, datetime(2009, 1, 1), None,),
+                        (0, AuditTypes.supplier_update, datetime(2010, 2, 3), None,),
+                        (1, AuditTypes.supplier_update, datetime(2010, 4, 9), None,),
+                        (2, AuditTypes.supplier_update, datetime(2010, 1, 1), None,),
+                        (1, AuditTypes.supplier_update, datetime(2010, 8, 8), None,),
+                    ),
+                    (  # series of req_cases for the above db scenario
```

Comment:
Oh - I was avoiding this because I was under the impression that it would 404 in such a case, but I think I must have been temporarily mad, because it doesn't 404 at all.

Maybe something else was doing it.

Yes, I will.